tasks/deploy/githubapp.py

```diff
@@ -0,0 +1,127 @@
+import base64
+import logging
+import os
+import time
+
+logger = logging.getLogger(__name__)
+logger.setLevel(os.environ.get('LOGGING_LEVEL', 'INFO'))
+
+# Defining a custom exception class to throw if the requests error out
+class GithubAppException(Exception):
+    pass
+
+
+class GithubApp:
+    def __init__(self):
+        self.key_b64 = os.environ['GITHUB_KEY_B64']
+        self.app_id = os.environ['GITHUB_APP_ID']
+        self.base_url = 'https://api.github.com'
+        self.installation_id = os.environ.get('GITHUB_INSTALLATION_ID', None)
+        if self.installation_id is None:
+            # Even if we don't know the installation id, there's an API endpoint to
+            # retrieve it, given the other credentials (app id + key).
+            self.installation_id = self.get_installation()
+
+    def gen_token_payload(self):
+        """
+        Generate the token needed to authenticate as a Github App.
+        """
+        # To get authenticated, you need to use a token with:
+        # - iat (creation_time)
+        # - exp (expiration time, here 5 minutes since we re-create a token every time we want to authenticate)
+        # - iss (the Github app id)
+        # and encode it with the private key that's associated with the app.
+        # See: https://docs.github.com/en/developers/apps/authenticating-with-github-apps#authenticating-as-a-github-app
+        epoch = int(time.time())
+        payload = {'iat': epoch, 'exp': epoch + (5 * 60), 'iss': self.app_id}
+        return payload
+
+    def get_headers(self):
+        """
+        Craft headers to make a request to the Github API as a Github App.
+        """
+        import jwt
+
+        token_payload = self.gen_token_payload()
+        bearer_token = jwt.encode(token_payload, base64.b64decode(self.key_b64), algorithm='RS256').decode()
+        headers = {
+            'Authorization': 'Bearer {}'.format(bearer_token),
+            'Accept': 'application/vnd.github.v3+json',
+        }
+        return headers
+
+    def make_request(self, endpoint, method='GET', payload=None):
```

Comment:
I guess ideally we would have a general class to interact with REST APIs but that looks like too much work. I am adding the comment in case someone thinks it's something that we should do in the future.